core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -802,16 +804,21 @@ private BlockMetadataManagerView getUpdatedView() {
    * @param blockId block id
    * @param oldLocation the source location of the block
    * @param newLocation new location to move this block
+   * @param wait lock the block until get a valid lock id
    * @return the resulting information about the move operation
    * @throws BlockDoesNotExistException if block is not found
    * @throws BlockAlreadyExistsException if a block with same id already exists in new location
    * @throws InvalidWorkerStateException if the block to move is a temp block
    */
   private MoveBlockResult moveBlockInternal(long sessionId, long blockId,
-      BlockStoreLocation oldLocation, BlockStoreLocation newLocation)
-          throws BlockDoesNotExistException, BlockAlreadyExistsException,
-          InvalidWorkerStateException, IOException {
-    long lockId = mLockManager.lockBlock(sessionId, blockId, BlockLockType.WRITE);
+      BlockStoreLocation oldLocation, BlockStoreLocation newLocation, boolean wait)
+      throws BlockDoesNotExistException, BlockAlreadyExistsException,
+      InvalidWorkerStateException, IOException {
+    long lockId = wait ? mLockManager.lockBlock(sessionId, blockId, BlockLockType.WRITE) :
```

Comment:
I feel the wait and retry logic should be handled by the client but not the server. @apc999 what do you think?